staging/src/github.com/openshift/oc/pkg/cli/version/version.go

```diff
@@ -1,25 +1,185 @@
 package version
 
 import (
+	"encoding/json"
+	"errors"
 	"fmt"
-	"io"
 
 	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
 
-	"k8s.io/apimachinery/pkg/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/tools/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/templates"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/oc/pkg/version"
+)
+
+type Version struct {
+	ClientVersion    *apimachineryversion.Info `json:"clientVersion,omitempty" yaml:"clientVersion,omitempty"`
```

Comment:
is there a struct upstream we can embed here instead of defining it completely ourself?

Reply:
I had it embedded at first, yes.  there is. I had it like that first, then changed it (to match what @soltysh was doing, lol)

Reply:
As we want to be compatible with kubectl's output, embedding the upstream one might make sense.

Reply:
I've embedded, to best of my ability, check it out please. : )

Reply:
Also added comments that the version cmd re-uses/copies logic from upstream in Complete and Run functions